app/services/bill_service.rb

```diff
@@ -26,7 +26,154 @@ def bills_by_id
     json = JSON.parse(response.body, symbolize_names: true)
   end
 
-  def conn
+
+# def rep_details(full_state, sponsors)
+#   representatives = {}
+#   sponsors.map do |rep|
+#     response = os_conn.get("?") do |request|
+#       request.params['name'] = rep
+#       request.params['include'] = "other_names"
+#       request.params['jurisdiction'] = full_state
+#     end
+#     json = JSON.parse(response.body, symbolize_names: true)
+#     if json[:results].empty? && rep.downcase.include?('committee')
+#       json = {results: [{id: 0, name: "#{rep}}"}], pagination: {per_page:10, page: 1, max_page: 1, total_items: 0}}
+#       representatives[:sponsor_details] = json
+#     elsif json[:results].empty?
+#       sleep(1.minute)
+#       response = os_conn.get("?") do |request|
+#         request.params['name'] = rep.split.last
+#         request.params['include'] = "other_names"
+#         request.params['jurisdiction'] = full_state
+#       end
+#       json = JSON.parse(response.body, symbolize_names: true)
+#       representatives[:sponsor_details] = json
+#       binding.pry
+#     end
+
+#     representatives[:sponsor_details] = json
+#   end
+# end
+
+def rep_details(full_state, sponsors)
+  cache_key = "rep_details/#{full_state}/#{sponsors}"
+  representatives = Rails.cache.fetch(cache_key) do
+    fetch_rep_details_from_api(full_state, sponsors)
+  end
+  representatives
+end
+
+  
+  private
+
+  def fetch_rep_details_from_api(full_state, sponsors)
+    representatives = {}
+    sponsors.each do |rep|
+      response = os_conn.get("?") do |request|
+        request.params['name'] = rep
+        request.params['include'] = "other_names"
+        request.params['jurisdiction'] = full_state
+      end
+      json = JSON.parse(response.body, symbolize_names: true)
+
+      if json[:results].empty? && rep.downcase.include?('committee')
+        json = { results: [{ id: 0, name: "#{rep}}"}], pagination: { per_page: 10, page: 1, max_page: 1, total_items: 0 } }
+        representatives[:sponsor_details] = json
+      elsif json[:results].empty?
+        sleep(1.minute)
+        response = os_conn.get("?") do |request|
+          request.params['name'] = rep.split.last
+          request.params['include'] = "other_names"
+          request.params['jurisdiction'] = full_state
+        end
+        json = JSON.parse(response.body, symbolize_names: true)
+        representatives[:sponsor_details] = json
+      end
+
+      representatives[:sponsor_details] = json
+    end
+
+    representatives
```

Comment:
same with this api fetch_rep_details_method -- we may want to delete since its being refactored/worked on another branch